mirrord-layer/src/pod_api.rs

```diff
@@ -48,106 +48,371 @@ impl Drop for EnvVarGuard {
+                        "containers": [
+                            {
+                                "name": "mirrord-agent",
+                                "image": agent_image,
+                                "imagePullPolicy": self.config.agent.image_pull_policy,
+                                "securityContext": {
+                                    "privileged": true,
+                                },
+                                "volumeMounts": [
+                                    {
+                                        "mountPath": runtime_data.socket_path,
+                                        "name": "sockpath"
+                                    }
+                                ],
+                                "command": agent_command_line,
+                                "env": [{"name": "RUST_LOG", "value": self.config.agent.log_level}],
+                            }
+                        ]
+                    }
+                }
+            }
+                }
+            ))?;
+        let job_api = self.get_agent_api();
+
+        job_api
+            .create(&PostParams::default(), &agent_pod)
+            .await
+            .map_err(LayerError::KubeError)?;
+
+        let params = ListParams::default()
+            .labels(&format!("job-name={}", mirrord_agent_job_name))
+            .timeout(60);
+
+        pod_progress.done_with("agent pod created");
+
+        let pod_progress = progress.subtask("waiting for pod to be ready...");
+
+        let pod_api: Api<Pod> = self.get_agent_api();
+
+        let stream = watcher(pod_api.clone(), params).applied_objects();
+        pin!(stream);
+
+        while let Some(Ok(pod)) = stream.next().await {
+            if let Some(status) = &pod.status && let Some(phase) = &status.phase {
+                        debug!("Pod Phase = {phase:?}");
+                    if phase == "Running" {
+                        break;
+                    }
+                }
+        }
+
+        let pods = pod_api
+            .list(&ListParams::default().labels(&format!("job-name={}", mirrord_agent_job_name)))
+            .await
+            .map_err(LayerError::KubeError)?;
+
+        let pod_name = pods
+            .items
+            .first()
+            .and_then(|pod| pod.metadata.name.clone())
+            .ok_or(LayerError::JobPodNotFound(mirrord_agent_job_name))?;
+
+        wait_for_agent_startup(&pod_api, &pod_name, "mirrord-agent".to_string()).await?;
+
+        pod_progress.done_with("pod is ready");
+
+        Ok(pod_name)
+    }
+
+    pub async fn create_agent(&self, connection_port: u16) -> Result<Portforwarder> {
+        let progress = TaskProgress::new("agent initializing...");
+        let agent_image = agent_image(&self.config);
+        let runtime_data = self.get_runtime_data().await?;
+        let connect_pod_name = if self.config.agent.ephemeral {
+            self.create_ephemeral_container_agent(
+                runtime_data,
```

Comment:
Nitpick, but please have the same parameter order for create_ephemeral_container_agent and create_job_pod_agent

Reply:
fixed